### PR TITLE
[Tune] Parametrize Cloud Syncing Frequency

### LINKED ISSUE #7900: [tune] parametrize sync-to-cloud period
As a background, our project interacts with tune (0.8.3) via `tune.run_experiments(<dict_from_flatfile>)`. We noticed that sometimes our trial checkpoints were not promptly syncing to our S3 account. Without hardcoding an S3 upload in our `tune.Trainable._save()` function, I uncovered 2 hardcoded parameters that sometimes prevented sync-to-cloud. For now here's our monkey-patch fix, although I propose tune developers to kindly consider native parametrization support for the future please :)

```python
def set_tune_checkpoint_period(checkpoint_period_sec):
    """
    Control delay between syncs to cloud (AWS S3), from hardcoded defaults.
    - ray.tune.tune.run(global_checkpoint_period): 10 seconds
    - ray.tune.syncer.SYNC_PERIOD: 300 seconds
    """
    import inspect  # noqa: E402
    import ray.tune  # noqa: E402
    argspecs = inspect.getfullargspec(ray.tune.tune.run)
    arg_idx = argspecs.args.index("global_checkpoint_period")
    mandatory_arg_len = len(argspecs.args) - len(argspecs.defaults)
    new_defaults = list(argspecs.defaults)
    new_defaults[arg_idx - mandatory_arg_len] = checkpoint_period_sec
    ray.tune.tune.run.__defaults__ = tuple(new_defaults)

    import ray.tune.syncer  # noqa: E402
    ray.tune.syncer.SYNC_PERIOD = checkpoint_period_sec


set_tune_checkpoint_period(1)
```

### PR DESCRIPTION
**Pull request by amogkam**
## Why are these changes needed?
The frequency of syncing trial results to the cloud was previously hard coded to 300 seconds. This PR allows this setting to be configurable via an environment variable.

## Related issue number
Closes #7900

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: doc/source/tune/_tutorials/tune-usage.rst [hunk 1]
```diff
@@ -396,6 +396,8 @@ If a string is provided, then it must include replacement fields ``{source}`` an
         sync_process = subprocess.Popen(sync_cmd, shell=True)
         sync_process.wait()
 
+By default, syncing occurs every 300 seconds. To change the frequency of syncing, set the ``TUNE_CLOUD_SYNC_PERIOD_SEC`` environment variable in the driver to the desired syncing period. Note that the ``global_checkpoint_period`` argument also affects the frequency of uploading.
```
Comment (richardliaw) [inline doc/source/tune/_tutorials/tune-usage.rst:399]:
```suggestion
By default, syncing occurs every 300 seconds. To change the frequency of syncing, set the ``TUNE_CLOUD_SYNC_S`` environment variable in the driver to the desired syncing period. Note that the ``global_checkpoint_period`` argument also affects the frequency of uploading.
```

Comment (amogkam) [reply doc/source/tune/_tutorials/tune-usage.rst]: Changed

File: doc/source/tune/_tutorials/tune-usage.rst [hunk 2]
```diff
@@ -396,6 +396,8 @@ If a string is provided, then it must include replacement fields ``{source}`` an
         sync_process = subprocess.Popen(sync_cmd, shell=True)
         sync_process.wait()
 
+By default, syncing occurs every 300 seconds. To change the frequency of syncing, set the ``TUNE_CLOUD_SYNC_S`` environment variable in the driver to the desired syncing period. Note that the ``global_checkpoint_period`` argument also affects the frequency of uploading.
```
Comment (richardliaw) [inline doc/source/tune/_tutorials/tune-usage.rst:399]: Can you say more about how global_checkpoint_period affects frequency of uploading?

File: python/ray/tune/syncer.py [hunk 3]
```diff
@@ -70,12 +77,12 @@ def __init__(self, local_dir, remote_dir, sync_client=NOOP):
...
-    def sync_up_if_needed(self):
-        if time.time() - self.last_sync_up_time > SYNC_PERIOD:
+    def sync_up_if_needed(self, sync_period):
+        if time.time() - self.last_sync_up_time > sync_period:
             self.sync_up()
 
-    def sync_down_if_needed(self):
-        if time.time() - self.last_sync_down_time > SYNC_PERIOD:
+    def sync_down_if_needed(self, sync_period):
+        if time.time() - self.last_sync_down_time > sync_period:
```
Comment (richardliaw) [inline python/ray/tune/syncer.py:85]: document args?

File: doc/source/tune/_tutorials/tune-usage.rst (hunk 2 above)
Comment (amogkam) [reply doc/source/tune/_tutorials/tune-usage.rst]: added some more clarification here

File: python/ray/tune/syncer.py (hunk 3 above)
Comment (amogkam) [reply python/ray/tune/syncer.py]: done

**Review by richardliaw — APPROVED** [review]

**Pull request merged**